src/main/java/com/github/smuddgge/leaf/MessageManager.java

```diff
@@ -56,8 +60,12 @@ public static UUID getLastMessaged(UUID player) {
      * @param message The instance of the message.
      * @return The component.
      */
-    public static Component convertMiniMessage(String message) {
-        return MiniMessage.miniMessage().deserialize(message);
+    public static Component convertAndParseMiniMessage(String message, @Nullable Player player) {
```

Comment:
I think it could be better to put the mini placeholders in the placeholder parser rather then convert method.

Reply:
Ohh i see now, it turnes it into components